### PR TITLE
 Support all float dtypes in `F.sigmoid_cross_entropy`

### PR DESCRIPTION
**Pull request by takagi**
This PR follows #4582 to support all float dtypes in `F.sigmoid_cross_entropy`.

I found that it has relatively large error in forward computation when it computes FP64 data with CuPy. It seems that [this line](https://github.com/takagi/chainer/blob/c910fd0db9640784da9262908c7e4bdbb83c60b5/chainer/functions/loss/sigmoid_cross_entropy.py#L56) with dtype argument would cause the error between numpy and CuPy. I will check it later.

### CODE REVIEW

**Comment by takagi** [comment]
I got the cause of the problem. When `dtype=numpy.float64` is supplied to `cupy.divide`, `cupy.true_divide`'s `bb->d` routine is actually selected for its operation in Py3, consequently the arguments are casted before division.

In numpy, the same problem had ever caused and seems to be fixed in https://github.com/numpy/numpy/pull/9469 .

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Fixed the large error problem with CuPy FP64.

File: chainer/functions/loss/sigmoid_cross_entropy.py [hunk 1]
```diff
@@ -52,8 +52,10 @@ def forward(self, inputs):
             count = max(1, len(x))
         self.count = count
 
+        # TODO(takagi): Fix to perform division in a specific dtype. See
+        # cupy/cupy#1534.
         return utils.force_array(
-            xp.divide(xp.sum(loss), self.count, dtype=x.dtype)),
+            xp.divide(xp.sum(loss), self.count).astype(x.dtype)),
```
[the remark this answers is not in the document]
Comment (takagi) [reply chainer/functions/loss/sigmoid_cross_entropy.py]: Fixed!

**Comment by takagi** [comment]
Jenkins, test this please.

File: tests/chainer_tests/functions_tests/loss_tests/test_sigmoid_cross_entropy.py [hunk 2]
```diff
@@ -13,32 +13,62 @@
...
-        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-2}
+            -1, 1, self.shape).astype(self.dtype)
+
+        if self.dtype == numpy.float16:
+            self.places = 2
+            self.check_backward_options = {
+                'dtype':numpy.float64, 'atol': 5e-2, 'rtol': 5e-2}
+            self.check_double_backward_options = {
+                'dtype': numpy.float64, 'atol': 5e-2, 'rtol': 5e-2}
+        elif
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/loss_tests/test_sigmoid_cross_entropy.py]: Please add colon.

Comment (takagi) [reply tests/chainer_tests/functions_tests/loss_tests/test_sigmoid_cross_entropy.py]: oh...

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**